src/pages/index.js

```diff
@@ -92,118 +171,117 @@ enableValidation(settings);
-  profileImagePopup.close();
+  profileImagePopup.renderLoading(true);
+  api
+    .updateProfileAvatar({ avatar: newImageData.image })
+    .then((res) => {
+      user.setUserAvatar({ avatar: res.avatar });
+      profileImagePopup.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      profileImagePopup.renderLoading(false);
+    });
 }
 
 // Confirm card deletion handler
-function handleDeleteClick(card) {
+function handleDeleteCard(card) {
   cardToDelete = card;
-  confirmDeletePopup.open();
+  deleteCardPopup.open();
 }
 
+// Like button handler
+const handleLikeCard = (card) => {
+  const cardId = card.getId();
+  api
+    .likeCard(cardId)
+    .then(() => {
+      card.flipLikeState();
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    });
+};
+
 function handleConfirmDelete() {
-  if (cardToDelete) {
-    cardToDelete.deleteCard();
-    cardToDelete = null;
-    confirmDeletePopup.close();
-  }
+  deleteCardPopup.renderLoading(true);
+  api
+    .deleteCard(cardToDelete.getId())
+    .then(() => {
+      cardToDelete.handleDeleteCard();
+      cardToDelete = null;
+      deleteCardPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error deleting card:", err);
+    })
+    .finally(() => {
+      deleteCardPopup.renderLoading(false);
+    });
 }
 
 // Full card image handler
 function handleCardImageClick(name, link) {
   imagePreviewPopup.open(name, link);
 }
 
-// ! INSTANTIATE CLASSES
-
-// Confirm card delete form
-const confirmDeletePopup = new Popup(
-  "#confirm-delete-modal",
-  handleConfirmDelete
-);
-
-// New profile image form
-const profileImagePopup = new PopupWithForm(
-  "#profile-image-modal",
-  handleProfileImageSubmit
-);
-
-// Edit profile form
-const profileEditPopup = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileEditSubmit
-);
-
-// Add card form
-const addCardPopup = new PopupWithForm(
-  "#add-card-modal",
-  handleAddCardSubmit,
-  // new handler to allow for saving of data on modal close
-  updateCardFormData
-);
-
-// Full image popup
-const imagePreviewPopup = new PopupWithImage("#card-image-modal");
-
-//Section
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (item) => {
-      section.addItem(createCard(item));
-    },
-  },
-  "#card-list"
-);
-section.renderItems();
-
-// User Info
-const user = new UserInfo({
-  title: ".profile__title",
-  subtitle: ".profile__subtitle",
-});
-
 // ! EVENT LISTENERS FOR POPUPS
 
 profileEditPopup.setEventListeners();
 addCardPopup.setEventListeners();
 imagePreviewPopup.setEventListeners();
 profileImagePopup.setEventListeners();
+deleteCardPopup.setEventListeners();
 
 // ! EVENT LISTENERS FOR BUTTONS
 
 profileEditButton.addEventListener("click", () => {
-  const { title, subtitle } = user.getUserInfo();
+  const userInput = user.getUserInfo();
+  profileTitle.value = userInput.title;
+  profileSubtitle.value = userInput.title;
   profileEditPopup.open({
```

Comment:
```js

  profileTitle.value = userInput.title;
  profileSubtitle.value = userInput.title;
```

You have this logics inside `profileEditPopup.open`


```js

  profileEditPopup.open({
    title: userInput.title,
    subtitle: userInput.subtitle,
  });
```


This inserts the data